### PR TITLE
Proteomics Changes

### PR DESCRIPTION
**Pull request by cb4ds**

### CODE REVIEW

File: R/init.R [hunk 1]
```diff
@@ -83,8 +82,8 @@ initDGEobj <- function(primaryAssayData,
     )
 
     if (!allowShortSampleIDs == TRUE) {
```
Comment (agenius-mohammed-ali) [inline R/init.R:84]: For variable or conditions that evaluates to Boolean value at the end to we need to state == TRUE/FALSE?

File: R/init.R [hunk 2]
```diff
@@ -83,8 +82,8 @@ initDGEobj <- function(primaryAssayData,
...
-        suppressWarnings(
-            test <- as.numeric(rownames(colData))
+        suppressWarnings({
+            test <- as.numeric(rownames(colData))}
         )
```
Comment (agenius-mohammed-ali) [inline R/init.R]: Do we need to pull this closing parentheses to the previous line as well to have` })` ?

File: R/init.R [hunk 3]
```diff
@@ -83,8 +82,8 @@ initDGEobj <- function(primaryAssayData,
...
 
         if (all(is.na(test)) == FALSE) {
```
Comment (agenius-mohammed-ali) [inline R/init.R:89]: For variable or conditions that evaluates to Boolean value at the end to we need to state == TRUE/FALSE?

File: R/init.R [hunk 4]
```diff
@@ -101,7 +100,8 @@ initDGEobj <- function(primaryAssayData,
 
     funArgs <- match.call()
 
-    result <- try(primaryAssayData <- as.matrix(primaryAssayData), silent = TRUE)
+    result <- try({primaryAssayData <- as.matrix(primaryAssayData)}, silent = TRUE)
+
     if (any(class(result) == "try-error"))
         stop("Couldn't coerce primaryAssayData to a numeric matrix!")
```
Comment (agenius-mohammed-ali) [inline R/init.R:106]: Need to add `if` body inside `{}`

File: R/init.R [hunk 5]
```diff
@@ -120,43 +120,42 @@ initDGEobj <- function(primaryAssayData,
...
-                      item = primaryAssayData,
-                      itemName = paste(primaryAssayName, "_orig", sep=""),
-                      itemType = paste(primaryAssayName, "_orig", sep=""),
-                      funArgs = funArgs,
-                      parent = "",
-                      init = TRUE
-    )
+                      item     = primaryAssayData,
+                      itemName = paste(primaryAssayName, "_orig", sep = ""),
+                      itemType = paste(primaryAssayName, "_orig", sep = ""),
```
Comment (agenius-mohammed-ali) [inline R/init.R:132]: we can use `paste0` instead of `paste`

File: R/init.R [hunk 6]
```diff
@@ -120,43 +120,42 @@ initDGEobj <- function(primaryAssayData,
...
-                      item = primaryAssayData,
+                      item     = primaryAssayData,
                       itemName = DGEobjDef$primaryAssayNames[[level]],
                       itemType = DGEobjDef$primaryAssayNames[[level]],
-                      funArgs = funArgs,
-                      parent = paste(primaryAssayName, "_orig", sep=""),
-                      init = TRUE
-    )
+                      funArgs  = funArgs,
+                      parent   = paste(primaryAssayName, "_orig", sep = ""),
```
Comment (agenius-mohammed-ali) [inline R/init.R:142]: we can use paste0 instead of paste

File: R/init.R [hunk 7]
```diff
@@ -167,40 +166,53 @@ initDGEobj <- function(primaryAssayData,
...
+                      item     = rowData,
                       itemName = itemName,
                       itemType = itemType,
-                      funArgs = funArgs,
-                      parent = parent)
+                      funArgs  = funArgs,
+                      parent   = parent)
 
     # Annotate the level
     dgeObj %<>% setAttributes(list(level = level))
```
Comment (agenius-mohammed-ali) [inline R/init.R:183]: Do we need to refactor `%<>%` expression?

File: R/init.R [hunk 8]
```diff
@@ -167,40 +166,53 @@ initDGEobj <- function(primaryAssayData,
...
+                gr <- do.call("GRanges", list(rowData))
+            }, silent = TRUE)
+
+            if (class(result) != "try-error") {
+                dgeObj <- addItem(dgeObj,
+                                  item     = gr,
+                                  itemName = "granges_orig",
+                                  itemType = "granges_orig",
+                                  funArgs  = funArgs,
+                                  parent   = paste(grparent, "_orig", sep = ""))
```
Comment (agenius-mohammed-ali) [inline R/init.R:201]: we can use paste0 instead of paste

File: R/init.R [hunk 9]
```diff
@@ -167,40 +166,53 @@ initDGEobj <- function(primaryAssayData,
...
-                          parent = paste(grparent, "_orig", sep = ""))
-
-        dgeObj <- addItem(dgeObj,
-                          item = gr,
-                          itemName = "granges",
-                          itemType = "granges",
-                          funArgs = funArgs,
-                          parent = grparent)
+    if (level %in% c("exon", "gene")) {
+        gr_success <- FALSE
```
Comment (agenius-mohammed-ali) [inline R/init.R]: can we use `genomic_ranges_success` instead of `gr_success` to enhance readability and self documentation?

File: R/reset.R [hunk 10]
```diff
@@ -22,53 +22,26 @@
...
-        rowData <- getItem(dgeObj, "isoformData_orig")
-    } else if ("exonData_orig" %in% names(dgeObj)) {
-        rowData <- getItem(dgeObj, "exonData_orig")
-    } else if ("proteinData_orig" %in% names(dgeObj)) {
-        rowData <- getItem(dgeObj, "proteinData_orig")
-    } else {
-        stop("Gene/isoform/exon/protein data not found")
-    }
+    level     <- attr(dgeObj, "level")
+    levelData <- getItem(dgeObj, paste(level, "Data_orig", sep = ""))
```
Comment (agenius-mohammed-ali) [inline R/reset.R:36]: we can use `paste0` instead of `paste`

File: R/reset.R [hunk 11]
```diff
@@ -22,53 +22,26 @@
...
-    }
+    design    <- getItem(dgeObj, "design_orig")
+    def       <- attr(dgeObj, "objDef")
 
-    if ("effectiveLength_orig" %in% names(dgeObj)) {
-        newObj <- addItem(newObj,
-                          item = dgeObj$effectiveLength_orig,
-                          itemName = "effectiveLength_orig",
-                          itemType = "effectiveLength_orig")
+    paData    <- getItem(dgeObj, paste(def$primaryAssayNames[[level]], "_orig", sep = ""))
```
Comment (agenius-mohammed-ali) [inline R/reset.R:41]: we can use `paste0` instead of `paste`

File: R/init.R (hunk 1 above)
Comment (cb4ds) [reply R/init.R:84]: No we don't however this is existing code.

File: R/init.R (hunk 2 above)
Comment (cb4ds) [reply R/init.R]: Either, style preference.

File: R/init.R (hunk 3 above)
Comment (cb4ds) [reply R/init.R:89]: Nope, but this is the existing code.

File: R/init.R (hunk 4 above)
Comment (cb4ds) [reply R/init.R:106]: Agreed, but this is existing code again.

File: R/init.R (hunk 5 above)
Comment (cb4ds) [reply R/init.R:132]: Yes, however I'm following the existing style :-)

File: R/init.R (hunk 6 above)
Comment (cb4ds) [reply R/init.R:142]: Yes, however I'm following the existing style :-)

File: R/init.R (hunk 7 above)
Comment (cb4ds) [reply R/init.R:183]: Not at this time

File: R/init.R (hunk 9 above)
Comment (cb4ds) [reply R/init.R]: yes!  will update

**Pull request merged**